routers/web/repo/issue.go

```diff
@@ -990,7 +993,11 @@ func NewIssuePost(ctx *context.Context) {
 	}
 
 	log.Trace("Issue created: %d/%d", repo.ID, issue.ID)
-	ctx.Redirect(ctx.Repo.RepoLink + "/issues/" + fmt.Sprint(issue.Index))
+	if ctx.FormBool("redirect_on_project") {
```

Comment:
Is is necessary to introduce `redirect_on_project`? Can we just check if `form.ProjectID > 0`?

Reply:
I don't think it's possible, when form is submitted we can't know if user click on "new issue" button from project view, maybe he creates issue from issue list view.

Reply:
On which page besides `projects/view.tmpl` the new issue link uses `project_id=xxx` in query string?

Reply:
No one, but when you submit the new issue form with a project, the project_id is also filled ?

Reply:
I haven't read the source code carefully, just a feeling and curiosity: if there is no project_id, how can the newly created issue be added into current project?

Reply:
There is two cases : 
* First, you want to create an issue from button on list issue view, after on the form you can assign issue to a project. In this case, we don't want to redirect on the project but the project_id is filled. 
* Second, you want to create an issue from the button on project view, you are redirected to the new issue form with the project automatically selected. In this case we want to redirect on the project when the form is submitted 
If the project is not selected when the form is submitted the user is redirected to issue view

Reply:
For the first case, the `project_id` is in POST body. So `ctx.Req.URL.Query().Get("project_id")` will work like `redirect_on_project`, Query().Get("project_id") only presents when the issue is created from the page view button.

For me, I prefer to use use existing and simple mechanism as long as it doesn't introduce ambiguity.

Reply:
Ok I didn't understand what do you mean, sorry. So I removed `redirect_on_project` in URL and I used `ctx.Req.URL.Query().Get("project")` to know we should redirect on project page  or not

Reply:
I wasn't saying clearly.  😂